src/vm.rs

```diff
@@ -405,26 +405,28 @@ impl<T: GuestStagePageTable> Vm<T, VmStateFinalized> {
     }
 
     /// Run this guest until an unhandled exit is encountered.
-    fn run_vcpu(&self, vcpu_id: u64) -> EcallResult<TvmCpuExitCode> {
+    fn run_vcpu(
+        &self,
+        vcpu_id: u64,
+        parent_vcpu: Option<&mut ActiveVmCpu<T>>,
+    ) -> EcallResult<TvmCpuExitCode> {
         // Take the vCPU out of self.vcpus, giving us exclusive ownership.
         let mut vcpu = self
             .vcpus
             .take_vcpu(vcpu_id)
             .map_err(|_| EcallError::Sbi(SbiError::InvalidParam))?;
         let exit_code = {
-            let mut vcpu = vcpu.lock();
+            let mut locked_vcpu = vcpu.lock();
+            // Activate this vCPU, saving the state of our parent vCPU.
+            let mut active_vcpu = locked_vcpu.activate(&self.vm_pages, parent_vcpu).unwrap();
```

Comment:
Is the unwrap Ok here or can this fail?

Reply:
`activate()` will only fail if there's a mismatch between the page owner ID of `self.vm_pages` and the vCPU, so it's guaranteed to succeed here. But I went ahead and changed it to check for the error since it's easy enough to bail out here and it's conceivable that we'd add something down the road that could make `activate()` fail for other reasons.